relayer-cli/src/commands/config/auto.rs

```diff
@@ -0,0 +1,95 @@
+use std::path::PathBuf;
+
+use abscissa_core::clap::Parser;
+use abscissa_core::{Command, Runnable};
+use chain_registry::{error::RegistryError, relayer_config::hermes_config};
+
+use crate::conclude::Output;
+
+use futures::{stream::FuturesUnordered, StreamExt};
+
+use ibc_relayer::config::{store, ChainConfig, Config};
+
+use tokio::runtime::Builder;
+
+#[derive(Clone, Command, Debug, Parser, PartialEq)]
+pub struct AutoCmd {
+    #[clap(
+        long = "path",
+        short = 'p',
+        required = true,
+        value_name = "PATH",
+        help_heading = "REQUIRED",
+        help = "Path to the configuration file"
+    )]
+    path: PathBuf,
+
+    #[clap(
+        long = "chains",
+        required = true,
+        multiple = true,
+        value_name = "CHAIN_NAME1 CHAIN_NAME2...",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chain to include in the config"
+    )]
+    chain_ids: Vec<String>,
+
+    #[clap(
+        long = "key-names",
+        required = true,
+        multiple = true,
+        value_name = "KEY_NAME_CHAIN1 KEY_NAME_CHAIN2...",
+        help_heading = "REQUIRED",
+        help = "key names to include in the config"
+    )]
+    key_names: Vec<String>,
+}
+
+async fn get_chain_configs(
```

Comment:
Might be a good idea to validate that `chain_ids.len() == key_names.len()` at the start of this function.

Reply:
Should I remove the check at line 68 to avoid checking twice the same condition ?

Reply:
Ah, missed that. Nevermind then 🙂